src/rust/ide/view/graph-editor/src/lib.rs

```diff
@@ -2426,6 +2317,8 @@ fn new_graph_editor(app:&Application) -> GraphEditor {
     // === Edge creation  ===
 
     on_new_edge    <- any(&output_down,&input_down);
+    let selection_mode = selection::get_mode(network,inputs);
+    keep_selection <- selection_mode.map(|t| *t != selection::Mode::Normal);
```

Comment:
I don't understand this part, and I haven't found usage of `keep_selection`.

Reply:
It is actually used in the line just below: `deselect_edges <- on_new_edge.gate_not(&keep_selection);`
This is related to https://github.com/enso-org/ide/issues/529 but was never finished, I think.